File: backend/app/schemas/analytics.py
from typing import List, Optional, Dict, Any
from pydantic import BaseModel, ConfigDict
from datetime import date, datetime

# Backward compatibility aliases
class ChartPoint(BaseModel):
    name: str
    value: float

class TimeSeriesPoint(BaseModel):
    date: str
    revenue: float
    cost: float
    profit: float

class ServiceStat(BaseModel):
    service_id: int
    name: str
    category: Optional[str] = None
    revenue: float
    profit: float
    request_count: int
    pending_count: int
    processing_count: int
    completed_count: int
    staff_in_charge: List[str]
    is_active: bool = True
    is_public: bool = True
    is_available: bool = True
    pending_weight: Optional[float] = 0.0
    pending_cartons: Optional[int] = 0
    completed_weight: Optional[float] = 0.0

class RevenueTrend(TimeSeriesPoint): # Keep internal consistency
    pass

class CategoryStat(ChartPoint): # Keep internal consistency
    pass

class AnalyticsSummary(BaseModel):
    total_revenue: float
    total_cost: float
    net_profit: float
    total_debt: float
    # Workload Stats
    total_requests_count: int
    pending_requests_count: int
    approved_requests_count: int
    processing_requests_count: int
    completed_requests_count: int
    pending_verifications_count: int
    # Charts
    revenue_trend: List[TimeSeriesPoint]
    revenue_by_service: List[ChartPoint]
    debt_by_vendor: List[ChartPoint]
    # Detailed Stats
    service_stats: List[ServiceStat]

class UnifiedTransaction(BaseModel):
    id: str
    date: datetime
    type: str # INCOME / EXPENSE / LIABILITY
    category: str
    reference: str
    external_reference: Optional[str] = None
    amount: float
    status: str
    
    # Detail fields for Deep Dive Modal
    payment_method: Optional[str] = None
    claimed_amount: Optional[float] = None
    claimed_currency: Optional[str] = None
    exchange_rate: Optional[float] = None
    proof_url: Optional[str] = None
    actor_name: Optional[str] = None
    notes: Optional[str] = None
    service_request_id: Optional[int] = None
    user_id: Optional[int] = None

    model_config = ConfigDict(from_attributes=True)

# --- NEW: Categorized Performance Schemas ---

class StaffCategoryRevenue(BaseModel):
    category: str
    revenue: float

class StaffPerformanceReport(BaseModel):
    staff_id: int
    full_name: str
    staff_category: str
    work_office: str
    revenue_generated: float
    category_breakdown: List[StaffCategoryRevenue]
    operation_count: int
    travel_ticket_costs: float
    fixed_costs: float
    net_profit: float 

class CategorizedReport(BaseModel):
    month: str
    year: str
    internal_affairs_pnl: float
    external_affairs_pnl: float
    staff_performance: List[StaffPerformanceReport]
    global_stats: Dict[str, Any]

File: frontend/apps/balaka-admin/lib/analytics.ts
import { fetchClient } from "@/core/api";
import { format } from "date-fns";

export interface ServiceStat {
    service_id: number;
    name: string;
    revenue: number;
    profit: number;
    request_count: number;
    category?: string;
    pending_count: number;
    processing_count: number;
    completed_count: number;
    staff_in_charge: string[];
    is_active: boolean;
    is_public: boolean;
    is_available: boolean;
    pending_weight?: number;
    pending_cartons?: number;
    completed_weight?: number;
}

export interface AnalyticsSummary {
    total_revenue: number;
    total_cost: number;
    net_profit: number;
    total_debt: number;
    total_requests_count?: number;
    pending_requests_count: number;
    approved_requests_count: number;
    processing_requests_count: number;
    completed_requests_count: number;
    pending_verifications_count: number;
    revenue_trend: { date: string; revenue: number; cost: number; profit: number }[];
    revenue_by_service: { name: string; value: number }[];
    debt_by_vendor: { name: string; value: number }[];
    service_stats: ServiceStat[];
}

export async function getAnalyticsSummary(
    startDate?: Date | null, 
    endDate?: Date | null,
    staffId?: number | null,
    serviceId?: number | null,
    category?: string | null,
    isPublic?: boolean | null
): Promise<AnalyticsSummary> {
    const params = new URLSearchParams();
    if (startDate) params.append("start_date", format(startDate, "yyyy-MM-dd"));
    if (endDate) params.append("end_date", format(endDate, "yyyy-MM-dd"));
    if (staffId) params.append("staff_id", staffId.toString());
    if (serviceId) params.append("service_id", serviceId.toString());
    if (category) params.append("category", category);
    if (isPublic !== undefined && isPublic !== null) params.append("is_public", isPublic.toString());

    const queryString = params.toString() ? `?${params.toString()}` : "";
    return fetchClient<AnalyticsSummary>(`/api/v1/analytics/summary${queryString}`);
}
